### PR TITLE
fix(demo): improve Makefile commands

### PR DESCRIPTION
**Pull request by jeqo**
Changes:

- Add replication factor variable
- Disable time-based retention when size-based topics are created.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -132,9 +136,17 @@ subprojects {
         testRuntimeOnly "org.slf4j:slf4j-log4j12:$slf4jVersion"
     }
 
-    tasks.named('test') {
+    test {
```
Comment (AnatolyPopov) [inline build.gradle:139]: Please read gradle guidlines https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_pitfalls
This should actually be applied everywhere and I was trying to do it before but maybe I missed some places.

**Review by AnatolyPopov — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (jeqo) [reply build.gradle]: Thanks, @AnatolyPopov! Wasn't aware of this. Gradle can be very tricky sometimes.

Though, this is probably referring to changes from #328. Applied the fixes there. 

I have removed the dependency on that PR as changes here can be applied individually. Have a look

File: demo/Makefile [hunk 2]
```diff
@@ -59,9 +63,11 @@ create_topic_no_ts_by_size:
 		kafka-topics \
 		--bootstrap-server $(bootstrap_server) \
 		--create \
+		--config retention.ms=-1 \
 		--config segment.bytes=$(segment) \
```
Comment (AnatolyPopov) [inline demo/Makefile:67]: What does `no_ts_by_size` mean? Is it by size or by time? From the code I can guess that it's by size, but the name is weird in that case. Same in other similarly named tasks

**Review by AnatolyPopov — CHANGES_REQUESTED** [review]

File: demo/Makefile (hunk 2 above)
Comment (jeqo) [reply demo/Makefile:67]: sure, applying suggestions. have a look.

**Review by AnatolyPopov — APPROVED** [review]
LGTM

**Pull request merged**